contribute/design/1.x/cstor-operator/doc.md

```diff
@@ -46,107 +46,154 @@ Following is the proposed CSPC schema in go struct:
 ``` go
 // CStorPoolCluster describes a CStorPoolCluster custom resource.
 type CStorPoolCluster struct {
-  metav1.TypeMeta   `json:",inline"`
-  metav1.ObjectMeta `json:"metadata,omitempty"`
-  Spec              CStorPoolClusterSpec   `json:"spec"`
-  Status            CStorPoolClusterStatus `json:"status"`
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              CStorPoolClusterSpec   `json:"spec"`
+	Status            CStorPoolClusterStatus `json:"status"`
+	VersionDetails    VersionDetails         `json:"versionDetails"`
 }
 
 // CStorPoolClusterSpec is the spec for a CStorPoolClusterSpec resource
 type CStorPoolClusterSpec struct {
-  // Pools is the spec for pools for various nodes
-  // where it should be created.
-  Pools []PoolSpec `json:"pools"`
+	// Pools is the spec for pools for various nodes
+	// where it should be created.
+	Pools []PoolSpec `json:"pools"`
+
+	// DefaultResources are the compute resources required by the cstor-pool
+	// container.
+	// If the resources at PoolConfig is not specified, this is written
+	// to CSPI PoolConfig.
+	DefaultResources *corev1.ResourceRequirements `json:"resources"`
+
+	// AuxResources are the compute resources required by the cstor-pool pod
+	// side car containers.
+	DefaultAuxResources *corev1.ResourceRequirements `json:"auxResources"`
+
+	// Tolerations, if specified, are the pool pod's tolerations
+	// If tolerations at PoolConfig is empty, this is written to
+	// CSPI PoolConfig.
+	Tolerations []corev1.Toleration `json:"tolerations"`
```

Comment:
Should be also be DefaultTolerations as there is a placeholder for Tolerations in poolConfig already?

Reply:
thanks.. will incorporate this.